extensions/librdkafka/PublishKafka.cpp

```diff
@@ -289,122 +356,309 @@ bool PublishKafka::configureNewConnection(const std::shared_ptr<KafkaConnection>
+    if (result != RD_KAFKA_CONF_OK) {
+      logger_->log_error("PublishKafka: configure request.required.acks error result [%s]", errstr);
+      return false;
+    }
+  }
+  value = "";
+  if (context->getProperty(RequestTimeOut.getName(), value) && !value.empty()) {
+    core::TimeUnit unit;
+    if (core::Property::StringToTime(value, valInt, unit) &&
+        core::Property::ConvertTimeUnitToMS(valInt, unit, valInt)) {
+      valueConf = std::to_string(valInt);
+      rd_kafka_topic_conf_set(topic_conf_, "request.timeout.ms", valueConf.c_str(), errstr.data(), errstr.size());
+      logger_->log_debug("PublishKafka: request.timeout.ms [%s]", valueConf);
+      if (result != RD_KAFKA_CONF_OK) {
+        logger_->log_error("PublishKafka: configure request.timeout.ms error result [%s]", errstr);
+        return false;
+      }
+    }
+  }
+  value = "";
+  if (context->getProperty(MessageTimeOut.getName(), value) && !value.empty()) {
+    core::TimeUnit unit;
+    if (core::Property::StringToTime(value, valInt, unit) &&
+        core::Property::ConvertTimeUnitToMS(valInt, unit, valInt)) {
+      valueConf = std::to_string(valInt);
+      rd_kafka_topic_conf_set(topic_conf_, "message.timeout.ms", valueConf.c_str(), errstr.data(), errstr.size());
+      logger_->log_debug("PublishKafka: message.timeout.ms [%s]", valueConf);
+      if (result != RD_KAFKA_CONF_OK) {
+        logger_->log_error("PublishKafka: configure message.timeout.ms error result [%s]", errstr);
+        return false;
+      }
+    }
+  }
+
+  rd_kafka_topic_t* topic_reference = rd_kafka_topic_new(conn->getConnection(), topic_name.c_str(), topic_conf_);
+  if (topic_reference == nullptr) {
+    rd_kafka_resp_err_t resp_err = rd_kafka_last_error();
+    logger_->log_error("PublishKafka: failed to create topic %s, error: %s", topic_name.c_str(), rd_kafka_err2str(resp_err));
+    return false;
+  }
+
+  // The topic took ownership of the configuration, we must not free it
+  confGuard.disable();
+
+  std::shared_ptr<KafkaTopic> kafkaTopicref = std::make_shared<KafkaTopic>(topic_reference);
+
+  conn->putTopic(topic_name, kafkaTopicref);
 
   return true;
 }
 
 void PublishKafka::onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session) {
-  logger_->log_trace("Enter trigger");
-  std::shared_ptr<core::FlowFile> flowFile = session->get();
+  // Check whether we have been interrupted
+  if (interrupted_) {
+    logger_->log_info("The processor has been interrupted, not running onTrigger");
+    context->yield();
+    return;
+  }
 
-  if (!flowFile) {
+  // Try to get a KafkaConnection
+  std::string client_id, brokers;
+  if (!context->getProperty(ClientName.getName(), client_id)) {
+    logger_->log_error("Client Name property missing or invalid");
+    context->yield();
+    return;
+  }
+  if (!context->getProperty(SeedBrokers.getName(), brokers)) {
```

Comment:
These should be moved to onschedule in scope of a follow-up ticket.

Reply:
Agreed.